contentcuration/contentcuration/static/js/edit_channel/models.js

```diff
@@ -292,6 +292,18 @@ var ContentNodeModel = BaseModel.extend({
     has_related_content: function(){
         return this.get('prerequisite').length || this.get('is_prerequisite_of').length;
     },
+    get_source_channel_id: function() {
+      var source_channel = this.get('source_channel');
+      return source_channel ? source_channel['id'] : 'unknown_channel_id';
+    },
+    get_source_channel_title: function() {
+      var source_channel = this.get('source_channel');
+      return source_channel ? source_channel['name'] : 'Unknown channel';
```

Comment:
Might be a good place to either send back empty string and say `{{#if channel.name}}{{channel.name}}{{else}}{{formatMessage (intlGet 'messages.recently_added')}}{{/if}}` in the front-end or just send back `_("Recently Added")` here. You will need to translate it at some point- here are some examples of how that's handled:

1. On backend:  Use `_("string")` for backend ([see here](https://github.com/learningequality/studio/blob/develop/contentcuration/contentcuration/forms.py#L35))
2. On frontend: Use `{{formatMessage (intlGet 'messages.message_name')}}`  (see [here](https://github.com/learningequality/studio/blob/develop/contentcuration/contentcuration/static/js/edit_channel/tree_edit/hbtemplates/container_area.handlebars#L42) for template, and [here](https://github.com/learningequality/studio/blob/develop/contentcuration/contentcuration/static/js/edit_channel/tree_edit/views.js#L21) for setting message)

Reply:
Done! Thank you for the detailed explanation!